### ISSUE TITLE
[Bug]: When searching by IP measurement on collections with disk indexes, the milvus service is down all the time

### ISSUE THREAD

**Issue by lbj1110** [milvus-io/milvus#27279]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.3.0
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 24c/256G
- GPU: 
- Others:
```

### Current Behavior

When searching by IP measurement on collections with disk indexes, the milvus service is down all the time

### Expected Behavior

search success

### Steps To Reproduce

```markdown
1. Use docker-compose to deploy the milvus service, version 2.3.0 
2. insert about 2 million data and build diskann index 
3. Perform IP metric search on the collection, milvus service directly goes offline after about 10s 
4. Re-pull milvus, retry NO.3 step, milvus is down all the time
```

**Comment by xiaofan-luan** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Environment
> ```
> - Milvus version:2.3.0
> - Deployment mode(standalone or cluster):standalone
> - MQ type(rocksmq, pulsar or kafka):
> - SDK version(e.g. pymilvus v2.0.0rc2):
> - OS(Ubuntu or CentOS): 
> - CPU/Memory: 24c/256G
> - GPU: 
> - Others:
> ```
> 
> ### Current Behavior
> When searching by IP measurement on collections with disk indexes, the milvus service is down all the time
> 
> ### Expected Behavior
> search success
> 
> ### Steps To Reproduce
> ```
> 1. Use docker-compose to deploy the milvus service, version 2.3.0 
> 2. insert about 2 million data and build diskann index 
> 3. Perform IP metric search on the collection, milvus service directly goes offline after about 10s 
> 4. Re-pull milvus, retry NO.3 step, milvus is down all the time
> ```
> 
> ### Milvus Log
> _No response_
> 
> ### Anything else?
> _No response_

please share your milvus log and your test code.
And by the way did you check you have enough memory for load 2 million data and your disk is NVMe?
Sata disk an normal ssd won't work with diskANN due to its high iops requirement

**Comment by xiaofan-luan** [comment]
you can use IVFSQ8 instead if you don't have enough memory

**Comment by lbj1110** [comment]
def range_search():
    from pymilvus import CollectionSchema, FieldSchema, Collection, connections, DataType
    import random
    connections.connect(host=setting.MILVUS_HOST, port=setting.MILVUS_PORT)

    # create collection
    dim = 512
    fields = [
        FieldSchema(name="uid", dtype=DataType.INT64, is_primary=True),
        FieldSchema(name="vectors", dtype=DataType.FLOAT_VECTOR, dim=dim)
    ]
    schema = CollectionSchema(fields=fields)

    collection = Collection("DataEngineTrainClipQ3Test", schema=schema)

    # insert data
    nb = 3
    vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]

    # search iterator
    search_params = {
        "metric_type": "IP",
        "params": {
            "radius": 0.3,
            "range_filter": 0.5,
            "k": 10,
            "search_list": 50,
        }
    }
    search_iterator = collection.search_iterator(vectors[:1], "vectors", search_params)

    page_idx = 0
    while True:
        res = search_iterator.next()  # 翻下一页
        if len(res) == 0:
            print("search iteration finished, close")
            search_iterator.close()  # close
            break
        print(len(res))
        print(res)
        # for i in range(len(res)):
        #     print(res[i])
        page_idx += 1
        print(f"page{page_idx}-------------------------")

**Comment by xiaofan-luan** [comment]
this is range search? 
how you evaluate result with range search?

**Comment by xiaofan-luan** [comment]
"params": {
        "radius": 0.3,
        "range_filter": 0.5,
        "k": 10,
        "search_list": 50,
    }

are to doing topK search, or range search?

you are asking for top 10 result with distance between 0.3 and 0.5

please read the document before you did test, thanks for  the understanding.

**Comment by xiaofan-luan** [comment]
/assign @yanliang567 
could we try this internally?
this might be related to diskANN index

**Comment by xiaofan-luan** [comment]
> def range_search(): from pymilvus import CollectionSchema, FieldSchema, Collection, connections, DataType import random connections.connect(host=setting.MILVUS_HOST, port=setting.MILVUS_PORT)
> 
> ```
> # create collection
> dim = 512
> fields = [
>     FieldSchema(name="uid", dtype=DataType.INT64, is_primary=True),
>     FieldSchema(name="vectors", dtype=DataType.FLOAT_VECTOR, dim=dim)
> ]
> schema = CollectionSchema(fields=fields)
> 
> collection = Collection("DataEngineTrainClipQ3Test", schema=schema)
> 
> # insert data
> nb = 3
> vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
> 
> # search iterator
> search_params = {
>     "metric_type": "IP",
>     "params": {
>         "radius": 0.3,
>         "range_filter": 0.5,
>         "k": 10,
>         "search_list": 50,
>     }
> }
> search_iterator = collection.search_iterator(vectors[:1], "vectors", search_params)
> 
> page_idx = 0
> while True:
>     res = search_iterator.next()  # 翻下一页
>     if len(res) == 0:
>         print("search iteration finished, close")
>         search_iterator.close()  # close
>         break
>     print(len(res))
>     print(res)
>     # for i in range(len(res)):
>     #     print(res[i])
>     page_idx += 1
>     print(f"page{page_idx}-------------------------")
> ```

what kind of disk you are using? Did you watch for memory usage and disk usage? especially iops

**Comment by lbj1110** [comment]
[logs.txt](https://github.com/milvus-io/milvus/files/12712691/logs.txt)

**Comment by lbj1110** [comment]
> > def range_search(): from pymilvus import CollectionSchema, FieldSchema, Collection, connections, DataType import random connections.connect(host=setting.MILVUS_HOST, port=setting.MILVUS_PORT)
> > ```
> > # create collection
> > dim = 512
> > fields = [
> >     FieldSchema(name="uid", dtype=DataType.INT64, is_primary=True),
> >     FieldSchema(name="vectors", dtype=DataType.FLOAT_VECTOR, dim=dim)
> > ]
> > schema = CollectionSchema(fields=fields)
> > 
> > collection = Collection("DataEngineTrainClipQ3Test", schema=schema)
> > 
> > # insert data
> > nb = 3
> > vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
> > 
> > # search iterator
> > search_params = {
> >     "metric_type": "IP",
> >     "params": {
> >         "radius": 0.3,
> >         "range_filter": 0.5,
> >         "k": 10,
> >         "search_list": 50,
> >     }
> > }
> > search_iterator = collection.search_iterator(vectors[:1], "vectors", search_params)
> > 
> > page_idx = 0
> > while True:
> >     res = search_iterator.next()  # 翻下一页
> >     if len(res) == 0:
> >         print("search iteration finished, close")
> >         search_iterator.close()  # close
> >         break
> >     print(len(res))
> >     print(res)
> >     # for i in range(len(res)):
> >     #     print(res[i])
> >     page_idx += 1
> >     print(f"page{page_idx}-------------------------")
> > ```
> 
> what kind of disk you are using? Did you watch for memory usage and disk usage? especially iops

hdd disk

**Comment by lbj1110** [comment]
> this is range search?
> 
> could you offer your milvus log?

yes, it is range search on collection with diskann

**Comment by xiaofan-luan** [comment]
hdd disk might be good anough for disk ANN index. 
it highly use ann and might introduce pressure.

2M data seems not huge, use HNSW index instead for now?

**Comment by xiaofan-luan** [comment]
seems to have too much thread waiting on IOs

**Comment by lbj1110** [comment]
> anyway let's verify this and make sure this is our side issue @yanliang567

This experiment is mainly to test whether the diskann index supports range search, and then we have about 6 billion data on the milvus deployed in the cluster has established a disk index, which requires the use of range search function, and the hardware resources are sufficient, including NVME and memory. This time is only to verify the feasibility of range search, so eager to get your help, thank you

**Comment by xiaofan-luan** [comment]
> > anyway let's verify this and make sure this is our side issue @yanliang567
> 
> This experiment is mainly to test whether the diskann index supports range search, and then we have about 6 billion data on the milvus deployed in the cluster has established a disk index, which requires the use of range search function, and the hardware resources are sufficient, including NVME and memory. This time is only to verify the feasibility of range search, so eager to get your help, thank you

DiskANN should support range search. but DiskANN currently only support nvme disk.

**Comment by xiaofan-luan** [comment]
we will do more test and make sure diskANN + rangesearch is working

**Comment by xiaofan-luan** [comment]
6 billion is actaully large, please contact us if you need extra help.

**Comment by lbj1110** [comment]
> 6 billion is actaully large, please contact us if you need extra help.

In fact, we also encountered a very difficult problem in the process of creating a disk index: the indexing process is very slow and the indexNode node is often restarted, do I need to open another issue? By the way, if the cluster network environment is not very stable, it is better to deploy services with docker-compose

**Comment by xiaofan-luan** [comment]
> > 6 billion is actaully large, please contact us if you need extra help.
> 
> In fact, we also encountered a very difficult problem in the process of creating a disk index: the indexing process is very slow and the indexNode node is often restarted, do I need to open another issue? By the way, if the cluster network environment is not very stable, it is better to deploy services with docker-compose

DiskANN index build may also need disk or many memory, did you see any index pod crash?

**Comment by lbj1110** [comment]
> > > 6 billion is actaully large, please contact us if you need extra help.
> > 
> > 
> > In fact, we also encountered a very difficult problem in the process of creating a disk index: the indexing process is very slow and the indexNode node is often restarted, do I need to open another issue? By the way, if the cluster network environment is not very stable, it is better to deploy services with docker-compose
> 
> DiskANN index build may also need disk or many memory, did you see any index pod crash?

The current hardware resources are about 2T memory and 35T nvme, which is enough for 6 billion 512-dimensional feature vectors? Index pods do crash frequently

**Comment by xiaofan-luan** [comment]
35T nvme disk should be good enough. 

how large is index pod? 

@liliu-z  could you help on following up this?

**Comment by yanliang567** [comment]
@elstic please keep an eye on this issue.

/assign @elstic

**Comment by liliu-z** [comment]
Can not reproduce locally
